### PR TITLE
sys-devel/bpf-toolchain: new package, add 20240407

### PR DESCRIPTION
**Pull request by thesamesam**
Heavily inspired by, and based on, dev-util/mingw64-toolchain. This provides the bpf-unknown-none toolchain for use with DTrace (to be packaged) and systemd.

Users may wish to use crossdev if they want to customise their toolchain - this is "just" a convenience option for e.g. aforementioned ebuilds to depend on and have things Just Work.

TODO:
* mingw64-toolchain has bin-symlinks custom-cflags debug +strip, do we want/need any of those?

Bug: https://bugs.gentoo.org/917228

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by thesamesam** [comment]
I'll seek review when TODO is addressed, but feel free to comment on the stuff not covered by that.

File: sys-devel/bpf-toolchain/bpf-toolchain-20240407.ebuild [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		emake -j1 MAKEINFO=: V=1 DESTDIR="${BPFT_D}" install
+
+		declare -f bpft-${id} >/dev/null && edo bpft-${id}
+
+		popd >/dev/null || die
+	}
+
+	# build with same ordering that crossdev would do
+	bpft-build binutils
+	bpft-build gcc
```
Comment (ionenwks) [inline sys-devel/bpf-toolchain/bpf-toolchain-20240407.ebuild:173]: If this doesn't need multi-stage bootstrapping, is it really necessary to be combined in 1 ebuild? Was under the impression the process was going to be more involved.

Arguably mingw64-toolchain could've split out binutils too, but still had to combine w/ runtime for the multi-stage builds.

Comment (ionenwks) [reply sys-devel/bpf-toolchain/bpf-toolchain-20240407.ebuild:173]: (not that I think of it as a problem if just want to keep things in one place in a similar manner)

**Comment by thesamesam** [comment]
Merged in e802488062f680f9a5ffe2a2ccb3bbefc3a8adb5 with some minor TODO left which I'll return to.

**Pull request closed without merging**